pipeline_dp/pipeline_operations.py

```diff
@@ -86,6 +90,52 @@ def group_by_key(self, col, stage_name: str):
     def filter(self, col, fn, stage_name: str):
         return col | stage_name >> beam.Filter(fn)
 
+    def filter_by_key(self, col, public_partitions, data_extractors, stage_name: str):
```

Comment:
Could you please implement filter_by_key in LocalOperations?

It will be very similar to BeamOperations, but no need to consider join case.

Reply:
Done.